sympy/assumptions/assume.py

```diff
@@ -117,6 +119,20 @@ def binary_symbols(self):
                 return i.binary_symbols
         return set()
 
+class PolyadicAppliedPredicate(AppliedPredicate):
+    def __new__(cls, predicate, args):
+        if predicate.arity != len(args):
+            raise TypeError("%s takes %d argument but %d were given" % (predicate, predicate.arity, len(args)))
+        args = Tuple(*[_sympify(a) for a in args])
+        return Boolean.__new__(cls, predicate, args)
+
+    @property
+    def args(self):
+        return self.arg
+
+    def _eval_ask(self, assumptions):
+        return self.func.polyadic_eval(self.arg, assumptions)
```

Comment:
Let's just make the multipledispatch thing an official way that things are done in ask, alongside (and probably to replace) the current class handler methods.